### ISSUE TITLE
hangs with: jack server not running or cannot be started

### ISSUE THREAD

**Issue by ekrohn** [The-Compiler/qutebrowser#465]
Start qutebrowser 0.1.2 (freshly downloaded) from shell prompt.
`:open www.discourse.org`
qutebrowser hangs. Not burning CPU time, window not refreshing, not killable with ^C, nor with kill (default signal) from a different window, but killable with kill -1. Output to terminal:
```
INFO: Request to www.google-analytics.com blocked by host blocker.
Cannot connect to server socket err = No such file or directory
Cannot connect to server request channel
jack server is not running or cannot be started
```
Note: pulseaudio is known to be broken on this debian testing machine, but that should not cause a browser to hang indefinitely.

**Comment by The-Compiler** [comment]
Sorry for not following up on this, I was a bit busy.

This output here:

```
Cannot connect to server socket err = No such file or directory
Cannot connect to server request channel
jack server is not running or cannot be started
```

is neither by qutebrowser nor by Qt. It smells like Qt is using some library or calling something which is not expected to hang, and that errors and hangs.

Do you have an idea how I could reproduce this so I can report it upstream? How is pulseaudio broken exactly? Why is jack mentioned in the error?

**Comment by ekrohn** [comment]
I'm not sure how much I can really tell you. I've had broken audio problems
on this Debian jessie machine for a while. Wish I could fix it. VLC also
hangs with this message:

[00000000021373a8] pulse audio output error: PulseAudio server connection failure: Connection refused

I can understand why an audio application would hang waiting for the
connection, but I was a bit surprised to hit it in a web browser.

Later....

I poked around and found that I had configured /etc/pulse/daemon.conf with:

daemonize = yes
system-instance = yes

I am thinking that I (mistakenly) did that 10 months ago to "fix" something
but I wonder if I just planted a time bomb that broke sound some time later.

Anyways, if you want to reproduce the problem, I think that does it (along
with killing pulseaudio and letting the system automatically start it again).

With the non-system-instance settings of:
; daemonize = no
; system-instance = no

qutebrowser 0.1.2 renders www.discourse.org fine with no hang. I'm happy.

--
Eric Krohn
